Adding new Staff Engineer role and modifying other areas accordingly

sabyz opened the pull request.

sabyz commented:
@thzinc Added another commit with 
- new description of our Org structure
- added Quality Assurance as a new distinction for now (fitting into engineer titles in the near future). 
- Engineering manager title since we already have current roles
- Engineering Management path 

Let me know what you think.

thzinc commented on README.md:
```diff
@@ -39,9 +44,10 @@ We have built this career framework to have two dimensions: Title and Distinctio
+| Senior Engineer | Their squad/ squad product area | Translates ideas into projects with discrete tasks and guides and unblocks others on the team. Collaborates with non-technical team members to give technical advice. | Has a consistent record of **very strong ownership** for their area, e.g. figuring out on-call schedules, establishing monitoring
+| Staff Engineer | Upto 2 squads/ multiple projects / squad product area(s) | Leads design and development of new product features/ technologies/ processes for their squad(s). Enables collaboration, processes and communication within and across squads and stakeholders.  | Exhibits the **highest level** of ownership across their squads and squad product area.  
+| Principal Engineer | Multiple squads/ all projects/ all product areas | Leads architecture of new systems/technologies/processes to improve the business. Communicates complex matters in understandable terms for a wide range of audiences. Implements best practices, provides governance over Engineering policies and procedures. | Exhibits ownership across the org - this person is a guardian of **G ...
+| Engineering Manager | Multiple squads/ all projects/ all product areas | Manages engineering projects, products, squads, operations, administration, SDLC, decision-making, provides governance over policies and best practices   | Exhibits **highest level** of ownership across their squads and squad product area(s)
```
I'm concerned about making `Engineering Manager` the highest title in this career framework. It reads like the pinnacle of engineering career development is management–something I disagree with.  Managing people and projects effectively is a set of skills that should be represented in a separate framework of titles.

I envision an engineering management track being a parallel set of titles to these, in which the highest title is probably your own: CTO. I think your description of an engineering manager informs this track well:

> Manages engineering projects, products, squads, operations, administration, SDLC, decision-making, provides governance over policies and best practices

A person on the team who is an "engineering manager" like this is the person who is championing process, putting feedback from retrospectives and post-mortems into effect, coordinating humans and making decisions when they are called for. I think there is room for variance in the seniority of a person on this track.  I think a minimal set of titles for such a track might look like:

* Manager
    * Scope of Influence: Their squad(s)
    * Primary Contributions
        * Coordinates work and process with squad leaders
        * Coordinates releases, on-call schedules, testing efforts
        * Contributes to budgeting for their squad's needs
    * Ownership: Owns process for a squad
* Senior Manager
    * Scope of Influence: Multiple squads
    * Primary Contributions
        * Coordinates work and process with other managers, squad leaders where necessary
        * Contributes to budgeting for their squads
    * Ownership: Owns the collaboration between their squads
* Chief Technology Officer
    * Scope of Influence: The entire engineering organization
    * Primary Contributions
        * Coordinates work and process with other managers, squad leaders where necessary
        * Plans financial resources for whole organization
        * Is a peer to and reports to the officers of the company
    * Ownership: all of engineering

sabyz replied:
@thzinc Engineering Manager is absolutely not the highest title. As we had discussed the seniority depends on the role level. I think the confusion is we don't have a level defined for any roles and these are parallel tracks as we had discussed and as I add Manager to the table assume that it's not sorted by Level. I think we can define and add levels (which I thought was an overkill for now) and simply included manager to cover for a role someone is already in (Leif). Other option is to add a separate table for title for Managers. It makes sense to have senior manager, director, CTO but I have not gotten to those yet. I'll make some edits and share with you.  We can discuss this further.

sabyz commented:
@thzinc I split up the tables for Engineer and Manager job titles and also added Levels to define seniority regardless of the career path within the org. That can help delineate reporting and compensation scales as well. Also, modified some of the verbiage for Contributions fusing with your copy keeping in mind current active roles. 

You will notice that a staff engineer can be senior to an L5 manager managing a single squad. 

Let me know your thoughts. We can setup an in-person meeting to go over this for the final edits once we have a fair draft.

sabyz commented:
Removed "levels" from previous feedback. This should now have all the changes discussed.
